src/main/java/io/cdap/plugin/salesforce/plugin/connector/SalesforceConnector.java

```diff
@@ -114,6 +114,13 @@ public BrowseDetail browse() throws IOException {
       for (int i = 0; i < dgr.getSobjects().length; i++) {
         String name = dgr.getSobjects()[i].getName();
         String label = dgr.getSobjects()[i].getLabel();
+        boolean queryable = dgr.getSobjects()[i].isQueryable();
+
+        // Continue in case of returning only queryable sObjects and the current sObject is non-queryable.
+        if (queryableOnly && !queryable) {
```

Comment:
nit: change the input paramter name to `onlyReturnQueryableObjects` so it's more semantically self-explanatory.

Also, th `queryable` variable should be `isQueryable`

```
if (onlyReturnQueryableObjects && !isQueryable) { ... }
```